darts/timeseries.py

```diff
@@ -829,6 +833,22 @@ def quantiles_df(self, quantiles: Tuple[float] = (0.1, 0.5, 0.9)) -> pd.DataFram
         # TODO: there might be a slightly more efficient way to do it for several quantiles at once with xarray...
         return pd.concat([self.quantile_df(quantile) for quantile in quantiles], axis=1)
 
+    def astype(self, dtype: Union[str, np.dtype]) -> 'TimeSeries':
+        """
+        Converts this TimeSeries to a new TimeSeries with desired dtype
+
+        Parameters
+        ----------
+        dtype
+            A NumPy dtype (np.float32 or np.float64)
+
+        Returns
+        -------
+        TimeSeries
+            A TimeSeries having the desired dtype.
+        """
+        return self.__class__(self._xa.astype(dtype))
+
```

Comment:
Nice one, I always wondered how we'd best convert dtypes for our torch models! 
Could we image adding option argument `dtype` to all of the `TimeSeries.from_*()` methods?
That way, users could avoid having to regenerate the time series to convert the values?

Reply:
Yes we could but not sure yet if it would be really needed so I'd rather leave that to a future PR - it's not too bad now:
`my_series = TimeSeries.from_csv(fname).astype(np.float32)` -vs-
`my_series = TimeSeries.from_csv(fname, dtype=np.float32)`